nutils/evaluable.py

```diff
@@ -764,10 +764,9 @@ def __get__(self, instance, owner):
         bound_orig = bind_orig(instance, owner)
         @functools.wraps(orig)
         def bound_method_wrapper(*args, **kwargs):
-          if kwargs.pop('skip_check', False):
-            return bound_orig(*args, **kwargs)
           res = bound_orig(*args, **kwargs)
-          assert res.dtype == instance.dtype, ((instance.dtype, res.dtype), instance, res)
+          if hasattr(instance, 'dtype'):
```

Comment:
`dtype` is not yet defined in initialisation of a `Piecewise`, see also the commit skip_check.